### PR TITLE
add WhoAmIResponse

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/13/
Test PASSed.

**Comment by mhilton** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/14/
Test PASSed.

File: csclient/csclient.go [hunk 1]
```diff
@@ -610,3 +610,13 @@ func (cs *Client) Login() error {
 	}
 	return nil
 }
+
+// WhoAmI returns the user and list of groups associated with the macaroon
+// used to authenticate.
+func (cs *Client) WhoAmI() (*params.WhoAmIResponse, error) {
+	var response params.WhoAmIResponse
+	if err := cs.Get("/whoami", &response); err != nil {
+		return nil, errgo.Notef(err, "cannot retrieve whoami response")
```
Comment (rogpeppe) [inline csclient/csclient.go]: Or just errgo.Mask(err); not sure the annotation adds much in this case.

**Comment by rogpeppe** [comment]
LGTM except that I think you'll need to split this into two, as discussed online.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/18/
Test PASSed.

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**